### DISCUSSION TITLE
Failed creating device object for moniker

### DISCUSSION THREAD

**Discussion by ab4728** [bonsai-rx/bonsai#1577] [Q&A]
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/e1646c87-46de-4cf2-b98e-52f8f3b9702e)

Does anyone have any advice on what to do with this error? 

![image](https://github.com/bonsai-rx/bonsai/assets/111145910/f597b1ae-8838-4776-8fb4-eb65ffa63390)

I will also include this error code. I'm especially confused by  "Spinnaker: GenICam::OutOfRangeException= Value = 1920 must be equal or smaller than Max = 1280." because the resolution of my behavioral tracking camera is not 1920 by any parameter. Notably, the camera in the fiber photometry system I'm running is 1920, but I have the index of videocapturedevice set to the right source, so I don't see how this could be the issue.

**Comment by ab4728** [comment]
I want my behavioral camera to be reading from pylon - not spinnaker, I don't know if there's a way in Bonsai to make this happen.

**Reply by bruno-f-cruz** [reply]
I am not sure I follow... The error you are getting in the terminal seems to come from the neurophotometrics node.

Regarding Pylon, I would simply use the PylonCapture node (the one you disabled at the bottom). But make sure you test it on a clean workflow, without any other knows to check if you get errors.

**Reply by bruno-f-cruz** [reply]
Can you do the following test?
1. Open a new workflow
2. Add a PylonCapture node
3. Select the Image property from this node

Does this simple workflow crash?

**Reply by ab4728** [reply]
That's a good question. I'm heading into work now so I'll try that in about 45 minutes.

**Reply by ab4728** [reply]
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/141503d0-93fe-4ebf-9fc6-f2e02bf1b810)

Yes. However, nothing appeared in the command window this time.

**Reply by ab4728** [reply]
If I run just videocapture device it doesn't crash, but I don't get any signal from the camera. This is the node we used in the past with this camera, but with the new neurophotometrics updates and our new computer, neurophotometrics recommended we start using pylon capture instead. 

![image](https://github.com/bonsai-rx/bonsai/assets/111145910/0eb5d7ef-d45f-4c2e-8d6b-0b23be9b4e04)

**Reply by bruno-f-cruz** [reply]
Can you do this:  https://github.com/orgs/bonsai-rx/discussions/1574#discussioncomment-7148832
And try the PylonCapture again?

**Reply by ab4728** [reply]
Sure. One moment.

**Reply by ab4728** [reply]
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/d9baef50-d5b8-47bc-8fba-0b9a2fc96d7c)

did a clean install of pylon and bonsai and getting this.

**Reply by bruno-f-cruz** [reply]
Have you installed the suggested dependency too? Install the 64bit and make sure you are running the 64bit version of bonsai too. Also, can you attach the bonsai.config of this system? This file can be found in the same folder as bonsai.exe.

**Reply by ab4728** [reply]
[Bonsai.txt](https://github.com/bonsai-rx/bonsai/files/12775252/Bonsai.txt)

By suggested dependency do you mean the OpenCV? My understanding was that was supposed to come with one of the packages. If there's something else I need to download I will.

How do I make sure I'm running the 64bit version?  In my file it appears I have two versions 
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/f00ddf8e-7920-4f96-9908-f2478a4a14e1)

and I also have an x86 version in my start menu 
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/ad996790-8970-4e15-82ca-0b518630a6e5)

**Reply by bruno-f-cruz** [reply]
The dependency I am referring to is mentioned in this post. Please install the 64bit one and run the bonsai.exe (not the bonsai32.exe)

**Reply by ab4728** [reply]
Oh you mean the Redistributables? Yes I have those and I just uninstalled and reinstalled them again and tried the bonsai.exe to test.

**Reply by ab4728** [reply]
Notably, I also get this error when I check installed packages.

![image](https://github.com/bonsai-rx/bonsai/assets/111145910/b1c2c8d0-d533-46f2-82f2-149a0cdab045)

**Reply by bruno-f-cruz** [reply]
Out of curiosity, could you try to install bonsai in the OS's root (i.e. C:/)? There is something funky going on with your user's path. I am wondering if that might be the problem.

**Reply by ab4728** [reply]
Sure. I also just found a remaining NuGet folder in that users folder that didn't get caught with the bonsai uninstallation so I deleted it. How do I control where bonsai installs to? 

(I appreciate the help so much.)

**Reply by ab4728** [reply]
Nvm I figured that part out. One moment.

**Reply by ab4728** [reply]
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/56e87773-e241-455b-b2bd-2c9ed693da8e)

So the unhandled exception error is gone. But the pylon capture error persists.

**Reply by ab4728** [reply]
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/d87babad-ae73-468d-9842-6927efc9b285)

Just as an update.

It is now running if I start a new file, running as admin. If I then load this "testingg" file, it will work. If I directly try to start it from the testing file, without running it on a new file first, it throws the exception.

**Reply by ab4728** [reply]
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/2c794911-cb10-41e2-97c2-2910deb78c0d)
If I try to load my workflow, disabling every other node. I get this other runtime error.

**Reply by ab4728** [reply]
Just tried deleting the parameter file and I'm back to this error

![image](https://github.com/bonsai-rx/bonsai/assets/111145910/c4616baa-d649-44f8-b045-63441fa15d0c)


I'm just including all of this information in case it helps any one make sense of all this.

**Reply by bruno-f-cruz** [reply]
Do you still have the bonsai installed in the app data folder? When you
open from file you are likely opening that executable that I told you I
suspected had something wrong with its path. That could explain why, if you
open the c:/ executable and then the workflow from it, it works.

**Reply by bruno-f-cruz** [reply]
How are you opening that file that crashes? Double click or are you opening the c:/bonsai/bonsai.exe , file->open

**Reply by ab4728** [reply]
![image](https://github.com/bonsai-rx/bonsai/assets/111145910/da4d9b7f-9620-4008-b43c-54c92fb7efcf)

Looks like there is another bonsai hidden in this path. Hopefully getting rid of this will fix something. One moment.

**Comment by bruno-f-cruz** [comment]
That’s just a shortcut. Please try the following:
Go your you c:/bonsai/bonsai.exe. Open bonsai. Click on the top left corner
File-> Open. Open your workflow. Click play. Does it still crash?

**Reply by ab4728** [reply]
No worries. And yes, it still crashes. Notably, every time I open a new file it immediately points it to the Gao Behavior room path, and the .nuget folder is getting saved in the Gao Behavior room path as well every time I reinstall.

**Reply by bruno-f-cruz** [reply]
Do you mind jumping onto [discord](https://discord.gg/Psa4TS75)? DM me we will take it from there.

**Reply by ab4728** [reply]
Sure, one last thing. If I start "camera capture" node, it will read the camera. Then if I delete that node and make a pylon capture node, it will run.

**Reply by ab4728** [reply]
Says invite link is expired

**Reply by bruno-f-cruz** [reply]
https://discord.gg/z3h4pWPs

**Reply by ab4728** [reply]
DM'd you.
